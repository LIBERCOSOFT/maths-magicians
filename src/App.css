* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.display,
.buttonContainer {
  width: 50%;
  margin: 0 auto;
}

.display {
  text-align: right;
  font-size: 3em;
  padding: 0.5em 0.1em;
  color: #fff;
  background-color: #858694;
  margin-top: 2em;
}

.buttonContainer > div {
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
}

.buttonContainer > div > button {
  padding: 2em;
  font-size: 1.5em;
  font-weight: 600;
  border: 0.1em solid #bababa;
}

.buttonContainer > div > button:hover {
  opacity: 0.7;
}

.buttonContainer > div > button:active {
  opacity: 0.5;
}

.buttonContainer > div > button:last-child {
  background-color: #f5913e;
}

.zero {
  grid-column: 1/3;
}

@media screen and (max-width: 875px) {
  .buttonContainer > div > button {
    padding: 0.5em;
    font-size: 1.5em;
    font-weight: 600;
  }

  .display {
    font-size: 1.7em;
  }
}

@media screen and (max-width: 428px) {
  .buttonContainer > div > button {
    padding: 0.5em;
    font-size: 1em;
    font-weight: 600;
  }

  .display {
    font-size: 1em;
  }
}

@media screen and (max-width: 300px) {
  .buttonContainer > div > button {
    padding: 0.5em;
    font-size: 0.5em;
    font-weight: 600;
  }
}
