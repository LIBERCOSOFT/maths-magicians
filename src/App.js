import React from 'react';
import { Router, Route, Routes } from 'react-router-dom';

import Calculator from './pages/Calculator';
import Home from './pages/Home';
import Quote from './pages/Quote';
import './App.css';
import Navbar from './components/Navbar';

const App = () => (
  <>
    <Navbar />
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quote" element={<Quote />} />
      </Routes>
    </Router>
  </>
);

export default App;
