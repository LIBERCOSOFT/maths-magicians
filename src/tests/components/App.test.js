import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import App from '../../App';

test('renders calculator component snapshot', () => {
  const component = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );
  expect(component).toMatchSnapshot();
});
