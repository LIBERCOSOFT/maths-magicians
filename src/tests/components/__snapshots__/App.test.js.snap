// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h1>
          Math Magicians
        </h1>
        <ul
          class="nav__links"
        >
          <li>
             
            <a
              aria-current="page"
              class="active__link"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
             
            <a
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
             
            <a
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="home__container"
      >
        <h2>
          Welcome to our page!
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sagittis nibh ut neque sollicitudin vestibulum. Maecenas varius leo quis sem efficitur pharetra. Curabitur posuere ex orci, nec ultricies augue aliquet sed. Pellentesque id arcu et quam ultrices congue nec at lectus. Vestibulum ultrices lacus malesuada aliquet tincidunt. Integer a fermentum quam. Etiam commodo vestibulum metus ut imperdiet. Vivamus congue augue a ex tristique auctor. Ut suscipit tellus nibh, ut porta lorem tincidunt sit amet. Integer ornare nibh mauris, ut pharetra urna mollis et. Phasellus commodo finibus ante sit amet malesuada. Suspendisse porttitor lorem libero, sed ullamcorper nisi pulvinar quis. Vestibulum eu arcu eget arcu aliquam porttitor. Nulla ut scelerisque mauris. Nunc quis urna vitae mauris tincidunt fermentum at vitae odio.
        </p>
        <p>
          Praesent vel bibendum augue. Nullam eu viverra augue. Quisque semper eu neque sit amet mollis. Vivamus at nisl congue, eleifend urna ac, posuere lacus. Quisque lobortis dapibus gravida. Curabitur at lectus porttitor, mollis ex non, sodales ipsum. Aenean tristique non mi eget elementum. Nunc ac vulputate eros, in euismod arcu.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h1>
        Math Magicians
      </h1>
      <ul
        class="nav__links"
      >
        <li>
           
          <a
            aria-current="page"
            class="active__link"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
           
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
           
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="home__container"
    >
      <h2>
        Welcome to our page!
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sagittis nibh ut neque sollicitudin vestibulum. Maecenas varius leo quis sem efficitur pharetra. Curabitur posuere ex orci, nec ultricies augue aliquet sed. Pellentesque id arcu et quam ultrices congue nec at lectus. Vestibulum ultrices lacus malesuada aliquet tincidunt. Integer a fermentum quam. Etiam commodo vestibulum metus ut imperdiet. Vivamus congue augue a ex tristique auctor. Ut suscipit tellus nibh, ut porta lorem tincidunt sit amet. Integer ornare nibh mauris, ut pharetra urna mollis et. Phasellus commodo finibus ante sit amet malesuada. Suspendisse porttitor lorem libero, sed ullamcorper nisi pulvinar quis. Vestibulum eu arcu eget arcu aliquam porttitor. Nulla ut scelerisque mauris. Nunc quis urna vitae mauris tincidunt fermentum at vitae odio.
      </p>
      <p>
        Praesent vel bibendum augue. Nullam eu viverra augue. Quisque semper eu neque sit amet mollis. Vivamus at nisl congue, eleifend urna ac, posuere lacus. Quisque lobortis dapibus gravida. Curabitur at lectus porttitor, mollis ex non, sodales ipsum. Aenean tristique non mi eget elementum. Nunc ac vulputate eros, in euismod arcu.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
